src/Compiler/Compile.hs

```diff
@@ -696,15 +763,30 @@ resolveModule term flags currentDir modules mimp
              --                            , loadedModules = allmods
              --                            }
              -- (loadedImp,impss) <- resolveImports term flags (dirname iface) loaded (map ImpCore (Core.coreProgImports (modCore mod)))
-             (imports,resolved1) <- resolveImportModules name term flags (dirname iface) modules (map ImpCore (Core.coreProgImports (modCore mod)))
+             (imports,resolved1) <- resolveImportModules compileTarget maybeContents name term flags (dirname iface) modules cachedModules (map ImpCore (Core.coreProgImports (modCore mod)))
              let latest = maxFileTimes (map modTime imports)
              -- trace ("loaded iface: " ++ show iface ++ "\n time: "  ++ show (modTime mod) ++ "\n latest: " ++ show (latest)) $ return ()
              if (latest >= modTime mod
                   && not (null source)) -- happens if no source is present but (package) depencies have updated...
-               then loadFromSource resolved1 root source -- load from source after all
-               else do liftIO $ copyPkgIFaceToOutputDir term flags iface (modCore mod) (modPackageQPath mod) imports
-                       let allmods = addOrReplaceModule mod resolved1
-                       return (mod{ modSourcePath = joinPath root source }, allmods)
+               then loadFromSource True resolved1 root source -- load from source after all
+               else 
+                let allmods = addOrReplaceModule mod resolved1
+                    result = (mod{ modSourcePath = joinPath root source }, allmods)
+                in case compileTarget of
+                  InMemory -> return result
```

Comment:
Here is the main difference in loadFromModule. We do not actually copy ifaces to output directory and we do not do code generation unless we need to (based on the last time the module was inferred and the last time it was written).